### PR TITLE
filters: add resetIdleTimer filter callback

### PR DESCRIPTION
**Pull request by goaway**
Description: This may be useful if a filter stops iteration for an extended period of time, since stream timeouts will still apply. The callback may be called periodically to continue to indicate "activity" on the stream. Note the final call to Envoy is commented out in this PR as we need the upstream change to land as well. Uncommenting and integration tests will come in a subsequent update.
Risk Level: Moderate
Testing: Local

Signed-off-by: Mike Schore <mike.schore@gmail.com>

### CODE REVIEW

File: library/common/extensions/filters/http/platform_bridge/filter.cc [hunk 1]
```diff
@@ -504,6 +514,18 @@ void PlatformBridgeFilter::resumeEncoding() {
   });
 }
 
+void PlatformBridgeFilter::resetIdleTimeout() {
+  ENVOY_LOG(trace, "PlatformBridgeFilter({})::resumeEncoding", filter_name_);
```
Comment (junr03) [inline library/common/extensions/filters/http/platform_bridge/filter.cc:518]:
```suggestion
  ENVOY_LOG(trace, "PlatformBridgeFilter({})::resetIdleTimer", filter_name_);
```

File: library/common/extensions/filters/http/platform_bridge/filter.cc [hunk 2]
```diff
@@ -504,6 +514,18 @@ void PlatformBridgeFilter::resumeEncoding() {
   });
 }
 
+void PlatformBridgeFilter::resetIdleTimeout() {
```
Comment (junr03) [inline library/common/extensions/filters/http/platform_bridge/filter.cc:517]:
```suggestion
void PlatformBridgeFilter::resetIdleTimer() {
```

File: library/common/extensions/filters/http/platform_bridge/filter.cc [hunk 3]
```diff
@@ -504,6 +514,18 @@ void PlatformBridgeFilter::resumeEncoding() {
...
+  ENVOY_LOG(trace, "PlatformBridgeFilter({})::resumeEncoding", filter_name_);
+
+  auto weak_self = weak_from_this();
+  dispatcher_.post([weak_self]() -> void {
+    if (auto self = weak_self.lock()) {
+      // Stream idle timeout is nondirectional.
+      //self->decoder_callbacks_->resetStreamIdleTimeout();
```
Comment (junr03) [inline library/common/extensions/filters/http/platform_bridge/filter.cc:524]:
```suggestion
      //decoder_callbacks_->resetStreamIdleTimeout();
```

File: library/common/extensions/filters/http/platform_bridge/filter.h [hunk 4]
```diff
@@ -62,6 +62,9 @@ class PlatformBridgeFilter final : public Http::PassThroughFilter,
   // This is a no-op if filter iteration is already ongoing.
   void resumeEncoding();
 
+  // Asynchronously reset the stream idle timeout. Does not affect other timeouts.
+  void resetIdleTimeout();
```
Comment (junr03) [inline library/common/extensions/filters/http/platform_bridge/filter.h:66]:
```suggestion
  void resetIdleTimer();
```

**Review by junr03 — COMMENTED** [review]
Lets also add PBF tests and swift/kotlin integration tests for the functionality

**Comment by goaway** [comment]
Updated with rename and added issue here for tests: https://github.com/envoyproxy/envoy-mobile/issues/1584

File: library/common/extensions/filters/http/platform_bridge/filter.cc (hunk 3 above)
Comment (goaway) [reply library/common/extensions/filters/http/platform_bridge/filter.cc]: Accessing through `self` is strictly necessary here, I think.

File: library/common/extensions/filters/http/platform_bridge/filter.cc [hunk 5]
```diff
@@ -504,6 +514,18 @@ void PlatformBridgeFilter::resumeEncoding() {
...
 }
 
+void PlatformBridgeFilter::resetIdleTimer() {
+  ENVOY_LOG(trace, "PlatformBridgeFilter({})::resetIdleTimer", filter_name_);
+
+  auto weak_self = weak_from_this();
+  dispatcher_.post([weak_self]() -> void {
+    if (auto self = weak_self.lock()) {
+      // Stream idle timeout is nondirectional.
+      // self->decoder_callbacks_->resetIdleTimer();
```
Comment (rebello95) [inline library/common/extensions/filters/http/platform_bridge/filter.cc:524]: Shouldn't this be uncommented?

Comment (goaway) [reply library/common/extensions/filters/http/platform_bridge/filter.cc:524]: It's commented because it relies on @junr03's PR landing.

**Review by rebello95 — APPROVED** [review]

**Pull request merged**